### PR TITLE
Fix bug in handling of ASG size params as pointers

### LINKED ISSUE #497: Output suggests creation of trillons of nodes.
**What happened?**
Created a cluster, with defaults, except for region, which I set to `eu-central-1`

**What you expected to happen?**
Everything went fine, except the output told me
```
[ℹ]  waiting for at least 824645529280 node(s) to become ready in "ng-40cbe42d"
```
, which gave be a brief shock. In the end, only 2 nodes have been created though.

**How to reproduce it?**
`eksctl create cluster --region eu-central-1`

**Anything else we need to know?**
Running on MacOS 10.13, downloaded eksctl from github.

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.1.20-rc.1"}
$ uname -a
Darwin jposchma-nb2.local 17.7.0 Darwin Kernel Version 17.7.0: Thu Dec 20 21:47:19 PST 2018; root:xnu-4570.71.22~1/RELEASE_X86_64 x86_64
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-13T23:16:58Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.5-eks-6bad6d", GitCommit:"6bad6d9c768dc0864dab48a11653aa53b5a47043", GitTreeState:"clean", BuildDate:"2018-12-06T23:13:14Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Logs**
```
$ eksctl create cluster --region eu-central-1
[ℹ]  using region eu-central-1
[ℹ]  setting availability zones to [eu-central-1a eu-central-1c eu-central-1b]
[ℹ]  subnets for eu-central-1a - public:192.168.0.0/19 private:192.168.96.0/19
[ℹ]  subnets for eu-central-1c - public:192.168.32.0/19 private:192.168.128.0/19
[ℹ]  subnets for eu-central-1b - public:192.168.64.0/19 private:192.168.160.0/19
[ℹ]  nodegroup "ng-40cbe42d" will use "ami-010caa98bae9a09e2" [AmazonLinux2/1.11]
[ℹ]  creating EKS cluster "hilarious-wardrobe-1548937947" in "eu-central-1" region
[ℹ]  will create 2 separate CloudFormation stacks for cluster itself and the initial nodegroup
[ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=eu-central-1 --name=hilarious-wardrobe-1548937947'
[ℹ]  creating cluster stack "eksctl-hilarious-wardrobe-1548937947-cluster"
[ℹ]  creating nodegroup stack "eksctl-hilarious-wardrobe-1548937947-nodegroup-ng-40cbe42d"
[ℹ]  --nodes-min=2 was set automatically
[✔]  all EKS cluster resource for "hilarious-wardrobe-1548937947" had been created
[✔]  saved kubeconfig as "/Users/JPoschma/.kube/config"
[ℹ]  nodegroup "ng-40cbe42d" has 0 node(s)
[ℹ]  waiting for at least 824645529280 node(s) to become ready in "ng-40cbe42d"
[ℹ]  nodegroup "ng-40cbe42d" has 2 node(s)
[ℹ]  node "ip-192-168-16-73.eu-central-1.compute.internal" is ready
[ℹ]  node "ip-192-168-80-88.eu-central-1.compute.internal" is ready
[ℹ]  kubectl command should work with "/Users/JPoschma/.kube/config", try 'kubectl get nodes'
[✔]  EKS cluster "hilarious-wardrobe-1548937947" in "eu-central-1" region is ready
```

### LINKED ISSUE #498: Scaling up cluster fails
**What happened?**
Scaling up cluster created by eksctl fails:
```
$ eksctl scale nodegroup --cluster=hilarious-wardrobe-1548937947 --name=ng-40cbe42d -N 4
[ℹ]  scaling nodegroup stack "eksctl-hilarious-wardrobe-1548937947-nodegroup-ng-40cbe42d" in cluster eksctl-hilarious-wardrobe-1548937947-cluster
[ℹ]  scaling nodegroup, desired capacity from  to 824638136896, max size from 2 to 824638136896
[✖]  unexpected status "UPDATE_ROLLBACK_IN_PROGRESS" while waiting for CloudFormation stack "eksctl-hilarious-wardrobe-1548937947-nodegroup-ng-40cbe42d" to reach "UPDATE_COMPLETE" status
[ℹ]  fetching stack events in attempt to troubleshoot the root cause of the failure
[ℹ]  AWS::CloudFormation::Stack/eksctl-hilarious-wardrobe-1548937947-nodegroup-ng-40cbe42d: UPDATE_ROLLBACK_IN_PROGRESS – "The following resource(s) failed to update: [NodeGroup]. "
[ℹ]  AWS::AutoScaling::AutoScalingGroup/NodeGroup: UPDATE_FAILED – "Encountered non numeric value for property MaxSize"
[ℹ]  AWS::CloudFormation::Stack/eksctl-hilarious-wardrobe-1548937947-nodegroup-ng-40cbe42d: UPDATE_IN_PROGRESS – "User Initiated"
[ℹ]  AWS::CloudFormation::Stack/eksctl-hilarious-wardrobe-1548937947-nodegroup-ng-40cbe42d: CREATE_COMPLETE
[ℹ]  AWS::AutoScaling::AutoScalingGroup/NodeGroup: CREATE_COMPLETE
[ℹ]  AWS::AutoScaling::AutoScalingGroup/NodeGroup: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::AutoScaling::AutoScalingGroup/NodeGroup: CREATE_IN_PROGRESS
[ℹ]  AWS::AutoScaling::LaunchConfiguration/NodeLaunchConfig: CREATE_COMPLETE
[ℹ]  AWS::AutoScaling::LaunchConfiguration/NodeLaunchConfig: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::AutoScaling::LaunchConfiguration/NodeLaunchConfig: CREATE_IN_PROGRESS
[ℹ]  AWS::IAM::InstanceProfile/NodeInstanceProfile: CREATE_COMPLETE
[ℹ]  AWS::IAM::InstanceProfile/NodeInstanceProfile: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::IAM::InstanceProfile/NodeInstanceProfile: CREATE_IN_PROGRESS
[ℹ]  AWS::IAM::Role/NodeInstanceRole: CREATE_COMPLETE
[ℹ]  AWS::EC2::SecurityGroupEgress/EgressInterClusterAPI: CREATE_COMPLETE
[ℹ]  AWS::EC2::SecurityGroupEgress/EgressInterCluster: CREATE_COMPLETE
[ℹ]  AWS::EC2::SecurityGroupEgress/EgressInterClusterAPI: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterCluster: CREATE_COMPLETE
[ℹ]  AWS::EC2::SecurityGroupEgress/EgressInterCluster: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterClusterAPI: CREATE_COMPLETE
[ℹ]  AWS::EC2::SecurityGroupEgress/EgressInterClusterAPI: CREATE_IN_PROGRESS
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterClusterCP: CREATE_COMPLETE
[ℹ]  AWS::EC2::SecurityGroupEgress/EgressInterCluster: CREATE_IN_PROGRESS
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterCluster: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterCluster: CREATE_IN_PROGRESS
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterClusterAPI: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterClusterCP: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterClusterAPI: CREATE_IN_PROGRESS
[ℹ]  AWS::EC2::SecurityGroupIngress/IngressInterClusterCP: CREATE_IN_PROGRESS
[ℹ]  AWS::EC2::SecurityGroup/SG: CREATE_COMPLETE
[ℹ]  AWS::EC2::SecurityGroup/SG: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::IAM::Role/NodeInstanceRole: CREATE_IN_PROGRESS – "Resource creation Initiated"
[ℹ]  AWS::IAM::Role/NodeInstanceRole: CREATE_IN_PROGRESS
[ℹ]  AWS::EC2::SecurityGroup/SG: CREATE_IN_PROGRESS
[ℹ]  AWS::CloudFormation::Stack/eksctl-hilarious-wardrobe-1548937947-nodegroup-ng-40cbe42d: CREATE_IN_PROGRESS – "User Initiated"
[✖]  failed to scale nodegroup for cluster "hilarious-wardrobe-1548937947", error waiting for CloudFormation stack "eksctl-hilarious-wardrobe-1548937947-nodegroup-ng-40cbe42d" to reach "UPDATE_COMPLETE" status: ResourceNotReady: failed waiting for successful resource state
```
**What you expected to happen?**
Cluster scaled up from 2 to 4 nodes.

**How to reproduce it?**
see What Happend?

**Anything else we need to know?**
Might be related to the broken node counts in #497

**Versions**
See #497

### PR DESCRIPTION
**Pull request by errordeveloper**
### Description

- Fix #497 
- Fix #498 

### Checklist
- [x] Code compiles correctly (i.e `make build`)
- [x] All tests passing (i.e. `make test`)

### CODE REVIEW

File: pkg/cfn/builder/nodegroup.go [hunk 1]
```diff
@@ -59,24 +59,31 @@ func (n *NodeGroupResourceSet) AddAllResources() error {
...
+		logger.Info("--nodes-min=%d was set automatically", *n.spec.MinSize)
 	} else if n.spec.DesiredCapacity != nil && *n.spec.DesiredCapacity < *n.spec.MinSize {
 		return fmt.Errorf("cannot use --nodes-min=%d and --nodes=%d at the same time", *n.spec.MinSize, *n.spec.DesiredCapacity)
 	}
 
 	// Ensure MaxSize is set, as it is required by the ASG cfn resource
 	if n.spec.MaxSize == nil {
 		if n.spec.DesiredCapacity == nil {
-			n.spec.MaxSize = n.spec.MinSize
+			if n.spec.MinSize == nil {
```
Comment (mumoshu) [inline pkg/cfn/builder/nodegroup.go]: This if-else seems redundant

File: pkg/cfn/manager/nodegroup.go [hunk 2]
```diff
@@ -158,35 +158,35 @@ func (c *StackCollection) ScaleNodeGroup(ng *api.NodeGroup) error {
 	currentMinSize := gjson.Get(template, minSizePath)
 
 	if ng.DesiredCapacity != nil && int64(*ng.DesiredCapacity) == currentCapacity.Int() {
-		logger.Info("desired capacity of nodegroup %q in cluster %q is already %d", ng.Name, clusterName, ng.DesiredCapacity)
+		logger.Info("desired capacity of nodegroup %q in cluster %q is already %d", ng.Name, clusterName, *ng.DesiredCapacity)
```
Comment (mumoshu) [inline pkg/cfn/manager/nodegroup.go]: Good catch! I found things like these in other places myself but my test wasn't enough. Thanks.

File: pkg/cfn/builder/nodegroup.go (hunk 1 above)
Comment (errordeveloper) [reply pkg/cfn/builder/nodegroup.go]: Yes, because it's actually wrong. Well spotted! 👍

File: pkg/cfn/builder/nodegroup.go [hunk 3]
```diff
@@ -76,7 +76,10 @@ func (n *NodeGroupResourceSet) AddAllResources() error {
 	if n.spec.MaxSize == nil {
 		if n.spec.DesiredCapacity == nil {
 			n.spec.MaxSize = n.spec.MinSize
+		} else {
+			n.spec.MaxSize = n.spec.DesiredCapacity
```
Comment (errordeveloper) [inline pkg/cfn/builder/nodegroup.go]: So all that was actually missing was this else clause.

**Pull request merged**